protobufs/livekit_metrics.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package livekit;
+option go_package = "github.com/livekit/protocol/livekit";
+option csharp_namespace = "LiveKit.Proto";
+option ruby_package = "LiveKit::Proto";
+
+
+message MetricsBatch {
+  repeated TimeSeriesMetric time_series = 1;
+  repeated AnnotationMetric annotations = 2;
+}
+
+message TimeSeriesMetric {
+  // Metric name, e.g., "speech_probability"
+  string label = 1;
+  // optional participant identity
+  string participant_identity = 2;
+  // optional track sid
+  string track_sid = 3;
+  // if track_sid is set, timestamp represents the RTP timestamp
+  int64 start_timestamp = 4;
+
+  oneof encoding {
+    DeltaEncoding delta_encoding = 5;
+    IntervalEncoding interval_encoding = 6;
+  }
+
+  message DeltaEncoding {
+    // protobuf encodes int32 as varint, delta times are usually small
+    repeated int32 delta_times = 1; // Delta time in milliseconds from the previous timestamp
+    repeated float values = 2;
+  }
+
+  message IntervalEncoding {
```

Comment:
Do we need this? Seems like a special case of delta encoding? Is it just meant to be a size optimization?

Reply:
if we're capturing data with high enough granularity that this sort of micro optimization makes sense processing and storing it will be prohibitively expensive

Reply:
I just removed the size implementation, as we will most likely always sample our real-time data at a maximum of 2 samples per second. (e.g speaking probability)